resources/aws_ec2.go

```diff
@@ -703,8 +742,128 @@ func (obj *AwsEc2Res) snsPostHandler(w http.ResponseWriter, req *http.Request) {
 		http.Error(w, "Invalid request method", http.StatusMethodNotAllowed)
 		return
 	}
-	post, _ := ioutil.ReadAll(req.Body)
-	if obj.debug {
-		log.Printf("%s: Post: %s", obj, string(post))
+	// decode json
+	decoder := json.NewDecoder(req.Body)
+	var p postData
+	err := decoder.Decode(&p)
+	if err != nil {
+		select {
+		case obj.awsChan <- &chanStruct{
+			err: errwrap.Wrapf(err, "error decoding post"),
+		}:
+		case <-obj.closeChan:
+		}
+		return
+	}
+	if p.Type == "SubscriptionConfirmation" {
+		csInput := &sns.ConfirmSubscriptionInput{
+			Token:    aws.String(p.Token),
+			TopicArn: aws.String(obj.snsTopicArn),
+		}
+		_, err := obj.snsClient.ConfirmSubscription(csInput)
+		if err != nil {
+			select {
+			case obj.awsChan <- &chanStruct{
+				err: errwrap.Wrapf(err, "error confirming subscription"),
+			}:
+			case <-obj.closeChan:
+			}
+			return
+		}
+		log.Printf("%s: Subscription Confirmed", obj)
```

Comment:
At this point, it's safe to say (once the cloudwatch code is added,) that we are officially watching the resource, so we can tell the engine we're running. We can do it here, or, to minimize side-effects, send a message to snsWatch and do obj.Running() there.